FunctionalTableDataTests/CellStyleTests.swift

```diff
@@ -13,6 +13,21 @@ class StyleTests: XCTestCase {
 	var cell: UITableViewCell!
 	var table: UITableView!
 	var style: CellStyle!
+
+	struct ColoredBackgroundProvider: BackgroundViewProvider {
+		let color: UIColor
+
+		public func backgroundView() -> UIView? {
+			let bgView = UIView()
+			bgView.backgroundColor = color
+			return bgView
+		}
```

Comment:
This will still create a new view every time we call this method. I thought this is what we are trying to avoid

Reply:
`CellStyle.configure(...)` calls this method, which is the correct time to create and apply the view to the cell. So instead of this:
```swift
let backgroundView = someView(...)
let cellStyle = CellStyle(backgroundView: backgroundView)
```
we can create a provider, which knows how to create the view, and init with that instead. Which would look like:

```swift
struct YellowProvider: BackgroundViewProvider {
        let backgroundColor: UIColor

        func backgroundView() -> UIView? {
                let view = UIView()
                view.backgroundColor = backgroundColor
                return view
        }
        ...
}

let provider = YellowProvider(backgroundColor: .red)
let cellStyle = CellStyle(backgroundViewProvider: provider)
```

This way we don't create the view in the builder (stopping ourselves from creating a view even if we're just to doing a comparison, not actually drawing anything), but instead we create it in `CellStye.configure(...)`.